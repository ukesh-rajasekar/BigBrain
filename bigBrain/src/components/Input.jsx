import React from 'react'
import PropTypes from 'prop-types'
import FormControl from 'react-bootstrap/FormControl';

function Input ({ name, placeholder, className, type, handleChange }) {
  return (
     
        <FormControl name={name} placeholder= {placeholder} className= {className} type={type} onChange = {(e) => handleChange(name, e.target.value)}/>
  )
}

Input.propTypes = {
  name: PropTypes.string,
  placeholder: PropTypes.string,
  className: PropTypes.string,
  type: PropTypes.oneOf(['text', 'number', 'password']),
  handleChange: PropTypes.func

}
Input.defaultProps = {
  type: 'text',
};

export default Input
