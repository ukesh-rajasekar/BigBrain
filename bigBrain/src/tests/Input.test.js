import React from 'react'
import { render, screen } from '@testing-library/react';
import { shallow, configure, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Input from '../components/Input';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });



describe('Render Input', () => {
  it('render correctly input component', () => {  
    const InputComponent = renderer.create(<Input />).toJSON();
    expect(InputComponent).toMatchSnapshot();
  });

  it('uses custom placeholder', () => {
    const className = 'login-input'
    const InputComponent = shallow(<Input className={className}/>);
    expect(InputComponent.props().className).toBe(className)
  })

  it('uses custom class name', () => {
    const placeholder = 'email'
    const InputComponent = shallow(<Input placeholder={placeholder}/>);
    expect(InputComponent.props().placeholder).toBe(placeholder)
  })

  it('uses custom type', () => {
    const type = 'password'
    const InputComponent = shallow(<Input type={type}/>);
    expect(InputComponent.props().type).toBe(type)
  })


  it('triggers onChange event handler when clicked', () => {
    const handleChange = jest.fn();
    shallow(<Input handleChange={handleChange}/>).simulate('change', { target: { name: 'email', value: 'a@b.com' } });
    expect(handleChange).toHaveBeenCalledTimes(1);
  })
 



  });
