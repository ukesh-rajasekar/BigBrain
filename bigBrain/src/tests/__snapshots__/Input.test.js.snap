// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Input render correctly input component 1`] = `
<input
  className="form-control"
  onChange={[Function]}
  type="text"
/>
`;
